### PR TITLE
Allow specifying additional extensions to load in Druid

### PR DESCRIPTION
**Pull request by labrenbe**
# Description

Closes https://github.com/stackabletech/druid-operator/issues/490

Currently the only way to enable additional extensions is to use config overrides potentially removing extension that were added to the extension load based by the operator

This PR adds a field `additionalExtensions` to `clusterConfig` containing a simple list of extension names. Extra configuration required by additional extensions is not in scope of this PR.

## Definition of Done Checklist

- Not all of these items are applicable to all PRs, the author should update this template to only leave the boxes in that are relevant
- Please make sure all these things are done and tick the boxes
 
```[tasklist]
# Author
- [ ] Changes are OpenShift compatible
- [X] CRD changes approved
- [X] CRD documentation for all fields, following the [style guide](https://docs.stackable.tech/home/nightly/contributor/docs-style-guide).
- [ ] Helm chart can be installed and deployed operator works
- [ ] Integration tests passed (for non trivial changes)
- [ ] Changes need to be "offline" compatible
```

```[tasklist]
# Reviewer
- [ ] Code contains useful comments
- [ ] Code contains useful logging statements
- [ ] (Integration-)Test cases added
- [ ] Documentation added or updated. Follows the [style guide](https://docs.stackable.tech/home/nightly/contributor/docs-style-guide).
- [ ] Changelog updated
- [ ] Cargo.toml only contains references to git tags (not specific commits or branches)
```

```[tasklist]
# Acceptance
- [ ] Feature Tracker has been updated
- [ ] Proper release label has been added
- [ ] [Roadmap](https://github.com/orgs/stackabletech/projects/25/views/1) has been updated
```

### CODE REVIEW

**Review by sbernauer — CHANGES_REQUESTED** [review]
As talked offline I pushed the experiment to `feat/additional-extensions-review`.
We also created https://github.com/kube-rs/kube/pull/1484

File: rust/crd/src/lib.rs [hunk 1]
```diff
@@ -300,6 +309,20 @@ pub struct DruidClusterConfig {
     pub listener_class: CurrentlySupportedListenerClasses,
 }
 
+/// TODO: Better docs
```
Comment (sbernauer) [inline rust/crd/src/lib.rs:312]: Could you please improve the docs a bit and link to https://github.com/kube-rs/kube/pull/1484? No need to write too much, will be removed in the future anyways

**Review by sbernauer — CHANGES_REQUESTED** [review]
Otherwise LGTM

File: rust/crd/src/lib.rs (hunk 1 above)
Comment (labrenbe) [reply rust/crd/src/lib.rs]: You're right, I've updated the docs comment

**Review by sbernauer — APPROVED** [review]
Many thanks!

**Pull request merged**